add console menu

stasevich741 opened the pull request.

stasevich741 committed:
add console menu

Maksim-maevsky commented on src/main/java/cinema/menu/Menu.java:
```diff
@@ -0,0 +1,37 @@
+        //      BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
+        Scanner reader = new Scanner(System.in);
+
+        System.out.println("===МЕНЮ===");
+        System.out.println("1. войти");
+        System.out.println("2. регистрация");
+        System.out.println("3. выход");
+
```
put this code in a separate method, and one sout will be enough.

stasevich741 replied:
fixed

stasevich741 committed:
add menu

stasevich741 committed:
add menu

stasevich741 committed:
add menu

stasevich741 committed:
add menu

Maksim-maevsky commented on src/main/java/cinema/entity/User.java:
```diff
@@ -13,9 +13,9 @@
         this.login = login;
         this.password = password;
-        this.userRole = userRole;
+        this.userRole = Role.USER;
```
will it only one role?

Maksim-maevsky commented on src/main/java/cinema/menu/Menu.java:
```diff
@@ -0,0 +1,99 @@
+    Scanner reader = new Scanner(System.in);
+    UserDao userDao = new UserDao();
+    FilmDao filmDao = new FilmDao();
+    UserActionMenu userActionMenu = new UserActionMenu();
```
private?

Maksim-maevsky replied:
where is construactor? I gues Scaner, UserDao, FilmDao and etc  You should create them to Main class.

Maksim-maevsky commented on src/main/java/cinema/menu/Menu.java:
```diff
@@ -0,0 +1,99 @@
+                } else if (command == 3) {
+                    showFilms();
+
+                } else if (command == 4) System.exit(0);
```
dont use exit method

stasevich741 replied:
fix... replace by `return`

Maksim-maevsky commented on src/main/java/cinema/menu/Menu.java:
```diff
@@ -0,0 +1,99 @@
+            while (true) {
+                int command = Integer.parseInt(reader.nextLine());
+
+                if (command == 1) {
```
what about switch

stasevich741 replied:
add switch

Maksim-maevsky commented on src/main/java/cinema/menu/Menu.java:
```diff
@@ -0,0 +1,99 @@
+    }
+
+    private void registration() {
+        System.out.println("введите логин");
```
convert to constant

stasevich741 replied:
done

Maksim-maevsky commented on src/main/java/cinema/menu/Menu.java:
```diff
@@ -0,0 +1,99 @@
+    private void registration() {
+        System.out.println("введите логин");
+        String login = reader.nextLine();
+        System.out.println("введите пароль");
```
convert to constant

Maksim-maevsky commented on src/main/java/cinema/menu/Menu.java:
```diff
@@ -0,0 +1,99 @@
+    }
+
+    private void authentication() {
+        System.out.println("введите логин");
```
convert to constant

Maksim-maevsky commented on src/main/java/cinema/menu/Menu.java:
```diff
@@ -0,0 +1,99 @@
+        String login = reader.nextLine();
+        System.out.println("введите пароль");
+        String password = reader.nextLine();
+        String sql_id_by_login_and_password = "select id from users where login='" + login + "' and password='" + password + "'";
```
try to folow  one style

Maksim-maevsky replied:
about naming

stasevich741 replied:
ok

Maksim-maevsky commented on src/main/java/cinema/menu/Menu.java:
```diff
@@ -0,0 +1,99 @@
+        String password = reader.nextLine();
+        String sql_id_by_login_and_password = "select id from users where login='" + login + "' and password='" + password + "'";
+        Long userId =0L;
+        try (Connection connection = ConnectionManager.get()) {
```
why is it there? it should be to repo and you need to call method from service

Maksim-maevsky commented on src/main/java/cinema/menu/Menu.java:
```diff
@@ -0,0 +1,99 @@
+        System.out.println("введите пароль");
+        String password = reader.nextLine();
+        String sql_id_by_login_and_password = "select id from users where login='" + login + "' and password='" + password + "'";
+        Long userId =0L;
```
gap

Maksim-maevsky commented on src/main/java/cinema/menu/UserActionMenu.java:
```diff
@@ -0,0 +1,27 @@
+
+public class UserActionMenu {
+
+    UserDao userDao = new UserDao();
```
use service method

Maksim-maevsky commented on src/main/resources/application.properties:
```diff
@@ -1,4 +1,3 @@
-db.driver=org.postgresql.Driver
```
why did you do this?

stasevich741 replied:
by this implementation app thows eror. then method works
```
java.lang.ClassNotFoundException: db.driver
    at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
    at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
    at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
    at java.base/java.lang.Class.forName0(Native Method)
    at java.base/java.lang.Class.forName(Class.java:315)
    at cinema.util.ConnectionManager.loadDriver(ConnectionManager.java:20)
    at cinema.util.ConnectionManager.<clinit>(ConnectionManager.java:15)
    at cinema.dao.FilmDao.findAll(FilmDao.java:19)
    at cinema.StartApp.main(StartApp.java:13)  
Film(id=1, name=saw, timeAndDate=2022-03-08T14:29:51.455397, ticket=100)
Film(id=2, name=saw, timeAndDate=2022-03-08T14:30:10.400193, ticket=100)
Film(id=3, name=saw, timeAndDate=2022-03-08T14:32:35.572262, ticket=100)
```

stasevich741 replied:
after throwing eror method works 
```
java.lang.ClassNotFoundException: db.driver
    at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
    at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
    at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
    at java.base/java.lang.Class.forName0(Native Method)
    at java.base/java.lang.Class.forName(Class.java:315)
    at cinema.util.ConnectionManager.loadDriver(ConnectionManager.java:20)
    at cinema.util.ConnectionManager.<clinit>(ConnectionManager.java:15)
    at cinema.dao.FilmDao.findAll(FilmDao.java:19)
    at cinema.StartApp.main(StartApp.java:13)  
Film(id=1, name=saw, timeAndDate=2022-03-08T14:29:51.455397, ticket=100)
Film(id=2, name=saw, timeAndDate=2022-03-08T14:30:10.400193, ticket=100)
Film(id=3, name=saw, timeAndDate=2022-03-08T14:32:35.572262, ticket=100)
```

stasevich741 committed:
add method userAuthenticationByLoginAndPassword in UserDao class
add private final to fields in Menu class
add menu's for user, admin, manager in UserActionMenu class

Maksim-maevsky commented on src/main/java/cinema/menu/Menu.java:
```diff
@@ -1,48 +1,52 @@
-    UserDao userDao = new UserDao();
-    FilmDao filmDao = new FilmDao();
-    UserActionMenu userActionMenu = new UserActionMenu();
+    private final Scanner reader;
```
private enough

Maksim-maevsky commented on src/main/java/cinema/menu/Menu.java:
```diff
@@ -57,43 +61,27 @@ private void showFilms() {
-                "2. регистрация" + "\n" +
-                "3. просмотреть мероприятия(фильмы)" + "\n" +
-                "4. выход");
+        System.out.println("===MENU===" + "\n" +
```
"===MENU===" + "\n" +
                "1. enter" + "\n" +
                "2. registration" + "\n" +
                "3. view events(movies)" + "\n" +
                "4. quit" - may be  is this  variable too?

Maksim-maevsky commented on src/main/java/cinema/menu/UserActionMenu.java:
```diff
@@ -6,22 +6,57 @@
+            System.out.println("===ADMIN_MENU===" + "\n" +
+                    "1. delete and edit user" + "\n" +
+                    "2. delete and edit events(movie)" + "\n" +
+                    "4. quit");
```
may be is this variable too?

Maksim-maevsky commented on src/main/java/cinema/menu/UserActionMenu.java:
```diff
@@ -6,22 +6,57 @@
 
         if (user.get().getUserRole() == Role.MANAGER) {
-            System.out.println("manager");
+            System.out.println("===MANAGER_MENU===" + "\n" +
+                    "1. edit events(movie)" + "\n" +
+                    "2. buy and return for user" + "\n" +
+                    "3. quit");
```
may be is this variable too?

Maksim-maevsky commented on src/main/java/cinema/menu/UserActionMenu.java:
```diff
@@ -6,22 +6,57 @@
+                    "2. buy ticket" + "\n" +
+                    "3. return ticket" + "\n" +
+                    "4. view bought tickets" + "\n" +
+                    "5. quit");
```
may be is this variable too?

stasevich741 committed:
add constant package
add menu package
add service package

Maksim-maevsky commented on src/main/java/cinema/constant/MenuSectionConst.java:
```diff
@@ -0,0 +1,34 @@
+package cinema.constant;
+
+public class MenuSectionConst {
+
```
is it constant class? it should be final, and final constructor

Maksim-maevsky commented on src/main/java/cinema/menu/MenuLogic.java:
```diff
@@ -0,0 +1,93 @@
+    private final AdminMenuImpl adminMenu;
+
+    private final FilmService filmService;
+    private final AuthService authService;
```
why all of this final?

Maksim-maevsky commented on src/main/java/cinema/menu/MenuLogic.java:
```diff
@@ -0,0 +1,93 @@
+        }
+    }
+
+    void userMenuSelection(Optional<User> user) {
```
selectUserMenu may be better?

Maksim-maevsky commented on src/main/java/cinema/menu/implementation/AdminMenuImpl.java:
```diff
@@ -0,0 +1,57 @@
+
+public class AdminMenuImpl implements Menu {
+
+    Scanner reader = new Scanner(System.in);
```
why is it not private?

Maksim-maevsky commented on src/main/java/cinema/menu/implementation/AdminMenuImpl.java:
```diff
@@ -0,0 +1,57 @@
+        }
+    }
+
+    private void editMovieByAdmin() {
```
I would change name?  without ByAdmin because

Maksim-maevsky commented on src/main/java/cinema/menu/implementation/AdminMenuImpl.java:
```diff
@@ -0,0 +1,57 @@
+
+    }
+
+    private void deleteMovieByAdmin() {
```
same

Maksim-maevsky commented on src/main/java/cinema/service/AuthService.java:
```diff
@@ -0,0 +1,33 @@
+    UserDao userDao = new UserDao();
+
+    public void registration() {
+        System.out.println(ASK_FOR_LOGIN);
+        String login = reader.nextLine();
+        System.out.println(ASK_FOR_PASSWORD);
+        String password = reader.nextLine();
+        User user = new User(login, password);
+        userDao.save(user);
```
separate modules of logic by empty line

Maksim-maevsky commented on src/main/java/cinema/service/AuthService.java:
```diff
@@ -0,0 +1,33 @@
+    }
+
+    public Optional<User> authentication() {
+        System.out.println(ASK_FOR_LOGIN);
+        String login = reader.nextLine();
+        System.out.println(ASK_FOR_PASSWORD);
+        String password = reader.nextLine();
```
same lines as above, what do you need to do? Think about constants and method

Maksim-maevsky commented on src/main/java/cinema/service/FilmService.java:
```diff
@@ -0,0 +1,22 @@
+
+public class FilmService {
+
+    FilmDao filmDao;
```
it last time when i write you about it, it should be private

Maksim-maevsky commented on src/main/java/cinema/menu/MenuLogic.java:
```diff
@@ -0,0 +1,93 @@
+
+    private final Scanner reader;
+
+    private final UserMenuImpl userMenu;
+    private final ManagerMenuImpl managerMenu;
+    private final AdminMenuImpl adminMenu;
```
Why are you added interface?)

Maksim-maevsky commented on src/main/java/cinema/menu/implementation/UserMenuImpl.java:
```diff
@@ -0,0 +1,53 @@
+                    case 4:
+                        viewBoughtTicketsByUser();
+                    case 5:
+                        System.out.println(MAIN_MENU);
```
you use this String several times add constant

stasevich741 committed:
implement methods from implementation package

stasevich741 committed:
implement methods from implementation package
add RoleActionConst class

Maksim-maevsky commented on src/main/java/cinema/menu/implementation/AdminMenuImpl.java:
```diff
@@ -64,28 +65,28 @@ void showTickets() {
         for (User user : users) {
             System.out.println(user);
-            menu();
+            showMenu();
```
will you show menu after each user?
